.github/workflows/unit_test.yml

```diff
@@ -0,0 +1,58 @@
+name: Run-Tests
+
+on:
+  pull_request:
+    branches: [develop, aws-poc]
+
+jobs:
+  scala-style:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-${{ hashFiles('**/pom.xml') }}
```

Comment:
What dos this line doing?

Reply:
caches the dependencies, this is the key it caches them by.

https://docs.github.com/en/free-pro-team@latest/actions/guides/caching-dependencies-to-speed-up-workflows#input-parameters-for-the-cache-action